improve installing kubeflow page

thesuperzapper opened the pull request:
*This PR replaces the previous https://github.com/kubeflow/website/pull/3381 PR which had a lot of around removing the "Kubeflow Version" from the table.*

The goal of this PR is to improve the experience of new users by improving the `Getting Started / Installing Kubeflow` page. We need this change because people are often overwhelmed by the current page's structure, or are confused when they see an inactive distribution for their chosen platform (and so think Kubeflow is not supported on their platform).

# Changes made to the "Installing Kubeflow" page

1. __Updated the section headings (for clarity):__
     - Added `How to install Kubeflow?` heading
     - Renamed `Install a packaged Kubeflow distribution` to `Packaged Distributions of Kubeflow`
     - Renamed `Install the Kubeflow Manifests manually` to `Raw Kubeflow Manifests`
2. __Splits the distributions up by "activity":__
     - "Active Distributions" - any distribution which has had an update within the last 6 months
     - "Legacy Distributions" - all other distributions (currently only, `Argoflow` and `Kubeflow on Azure`)
3. __Improves table formatting:__
     - Combines the distribution "links" into a single column, this is because most distributions now only have a single link for their external websites, after https://github.com/kubeflow/website/issues/3074
     - Added bullet points to "platforms" (if more than one is supported)
4. __Improved how we represent supported Kubeflow versions:__
     - Add a citation (if available) for the latest supported Kubeflow version
     - Marked distributions which do not use Kubeflow's version scheme as "proprietary" (only Arrikto for now)
5. __Corrected distribution names:__
     - Renamed `Kubeflow on OpenShift` to `Open Data Hub`
     - Renamed `Kubeflow on Oracle Container Engine for Kubernetes` to `Kubeflow on Oracle Cloud Infrastructure`
4. __Makes the other text on the page more clear:__
     - Added info about the future "conformance program", linked to https://github.com/kubeflow/kubeflow/issues/6485
     - Clarified how a user can install the raw manifests, and that this is an advanced use-case

# Screenshot (NEW)

![Installing-Kubeflow-Kubeflow](https://user-images.githubusercontent.com/5735406/215853947-210fcdc5-567f-431e-9a44-5cb27a05f015.png)

# Screenshot (Old)

![Installing-Kubeflow-Kubeflow](https://user-images.githubusercontent.com/5735406/215854056-1bff5be2-960e-4bed-9260-46e5b70f3cea.png)

thesuperzapper commented:
@jbottum @DomFleischmann @annajung @zijianjoy @surajkota I would greatly appreciate your review, hopefully, this PR will be less controversial than https://github.com/kubeflow/website/pull/3381!

---

I believe these changes are really important to improve the experience of new users. I understand there are lots of small changes (which might make it harder to review), but I recommend looking at the new page in aggregate to see how much more clear it is:
- https://deploy-preview-3440--competent-brattain-de2d6d.netlify.app/docs/started/installing-kubeflow/

Talador12 commented:
/lgtm

annajung commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,148 +9,242 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
-<b>See the table below for a list of options and links to documentation:</b>
+### Active Distributions
+
+The following table lists <b>active distributions</b> that have <b>had a recent release</b> (within the last 6 months).
```
How about 12 months? especially with the [1.7 release timeline](https://github.com/kubeflow/community/tree/master/releases/release-1.7) starting early September and plan to release by March, by the time 1.7 is released, every distribution would be "legacy" if not updated to latest version.

thesuperzapper replied:
> How about 12 months? especially with the [1.7 release timeline](https://github.com/kubeflow/community/tree/master/releases/release-1.7) starting early September and plan to release by March, by the time 1.7 is released, every distribution would be "legacy" if not updated to latest version.

@annajung We probably should aim to push releases more frequently, lol!

I don't mind updating it to say 12 months, but there is a risk that 12 months of inactivity becomes acceptable (once we are releasing more frequently) if that's what the website says.

jbottum commented:
I propose that the Distribution owners need to comment.   Since this represents their work, I believe they should decide and agree on this format.

thesuperzapper commented:
> I propose that the Distribution owners need to comment. Since this represents their work, I believe they should decide and agree on this format.

I think if after 2 weeks there is a general agreement we can probably go ahead and merge without explicitly waiting for each distribution. 

But we should make sure the distributions are aware of this PR so they are across it, are you willing to tag someone you know who is active for each active distribution? (Sorry that its a bit tedious).

thesuperzapper commented:
> I propose that the Distribution owners need to comment. Since this represents their work, I believe they should decide and agree on this format.

@jbottum I think if after 2 weeks there is a general agreement we can probably go ahead and merge without explicitly waiting for each distribution. 

But we should make sure the distributions are aware of this PR so they are across it, are you willing to tag someone you know who is active for each active distribution? (Sorry that its a bit tedious).

zijianjoy commented:
/lgtm

### Name

I am not sure about the following two changes, please confirm with distribution owners:

- Renamed `Kubeflow on OpenShift` to `Open Data Hub`
- Renamed `Kubeflow on Oracle Container Engine for Kubernetes` to `Kubeflow on Oracle Cloud Infrastructure`

### Platform

For Charmed Kubeflow, the `platform` is changed from `Conformant Kubernetes` to `All Conformant Kubernetes`. Is it intended?

For Nutanix, `Nutanix Karbon` is renamed to `Nutanix Kubernetes Engine`. Can we confirm with distribution owner?

StefanoFioravanzo commented:
@thesuperzapper Thanks for driving this refresh to the installation page. Can you please revert the Kubeflow version for "Arrikto Enterprise Kubeflow" and "Arrikto Kubeflow as a Service" to 1.5, please?

thesuperzapper commented:
> @thesuperzapper Thanks for driving this refresh to the installation page. Can you please revert the Kubeflow version for "Arrikto Enterprise Kubeflow" and "Arrikto Kubeflow as a Service" to 1.5, please?

@StefanoFioravanzo thanks, I have made this change in https://github.com/kubeflow/website/pull/3440/commits/484860abe219743bbb0d0dcdf98c5466bfd20f82, is everything else OK on your end now?

thesuperzapper commented:
> ### Name
> I am not sure about the following two changes, please confirm with distribution owners:
> 
> * Renamed `Kubeflow on OpenShift` to `Open Data Hub`
> * Renamed `Kubeflow on Oracle Container Engine for Kubernetes` to `Kubeflow on Oracle Cloud Infrastructure`

@zijianjoy I have reverted these changes in https://github.com/kubeflow/website/pull/3440/commits/734d25b2ae80e8e002354b69a5a85c15ed91ecb4, in the interest of getting this merged

> ### Platform
> For Charmed Kubeflow, the `platform` is changed from `Conformant Kubernetes` to `All Conformant Kubernetes`. Is it intended?

I think this means the same thing and is slightly more clear, so I don't think it's controversial.

> For Nutanix, `Nutanix Karbon` is renamed to `Nutanix Kubernetes Engine`. Can we confirm with distribution owner?

I updated this because `Nutanix Karbon` was renamed to `Nutanix Kubernetes Engine`, see from [their official docs which say "formally Karbon"](https://portal.nutanix.com/page/documents/details?targetId=Nutanix-Kubernetes-Engine-v2_7:Nutanix-Kubernetes-Engine-v2_7).

thesuperzapper commented:
@zijianjoy as long as you agree with my comment in https://github.com/kubeflow/website/pull/3440#issuecomment-1433972002, I think we are all good now?

StefanoFioravanzo commented:
@thesuperzapper Thanks for the change

/lgtm

jbottum commented:
@johnugeorge @Tomcli @annajung @DnPlas This PR has been open for a while and @thesuperzapper has been patient and persistent on finishing it.   I would like to ask that your team please reviews and comments and lgtm on this PR before Feb 23.   Thanks!    Note - No response will be assumed to be a looks good to me.

annajung commented:
cc @liuqi to review for VMware distribution

surajkota commented:
thanks for the updates. Minor changes for AWS distribution

Please change the hard coded Kubeflow version to short code

`{{% aws/latest-version %}}`

and update the release notes link to be generic - https://github.com/awslabs/kubeflow-manifests/releases

surajkota commented on content/en/docs/started/installing-kubeflow.md:
```diff
@@ -9,148 +9,242 @@ Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it prov
+## Raw Kubeflow Manifests
+
+{{% alert title="Warning" color="warning" %}}
+This method is for advanced users.
```
I will disagree and commit with this statement since this PR does not change the wording.

Kubeflow should be easy to install even if not using a distribution and it also sets a low bar for the community provided manifests & installation instructions.

thesuperzapper replied:
@surajkota I think this is a discussion for another time. There has always been significant debate about what sort of support should be provided for users of the raw manifests.

However, as it currently stands, we do not offer any environment-specific support, and so we must include a disclaimer that this is for advanced users.

thesuperzapper replied:
> Please change the hard coded Kubeflow version to short code so we can update it when distribution is released
> 
> `{{% aws/latest-version %}}`
> 
> and update the release notes link to be generic - https://github.com/awslabs/kubeflow-manifests/releases

@surajkota I have incorporated your requeted changes in commit https://github.com/kubeflow/website/pull/3440/commits/dba927df670be0ac152c2237f9cae320d35fdc68

thesuperzapper commented:
> Please change the hard coded Kubeflow version to short code so we can update it when distribution is released
> 
> `{{% aws/latest-version %}}`
> 
> and update the release notes link to be generic - https://github.com/awslabs/kubeflow-manifests/releases

@surajkota I have incorporated your requeted changes in commit https://github.com/kubeflow/website/pull/3440/commits/dba927df670be0ac152c2237f9cae320d35fdc68

surajkota commented:
/lgtm
AWS

jbottum commented:
Hi Folks, I propose that this has been open for review, we have / lgtms and we should merge this week.   Do we have an approve ?

zijianjoy commented:
/approve

I think we have obtained a few LGTMs which allow us to move forward with merging this change. 
To distribution owners: Feel free to make subsequent PRs if this change is not to your satisfaction.

Thank you @thesuperzapper for improving the Installing Kubeflow page! It is the front page of Kubeflow and it is important to guide users with the latest information.

julioo commented:
/lgtm
OCI
